src/Provider/Pool.php

```diff
@@ -127,16 +127,18 @@ public function hasContext(string $name): bool
     }
 
     /**
+     * @throws \LogicException
+     *
      * @phpstan-return array{
      *     providers: string[],
      *     formats: array<string, FormatOptions>,
      *     download: array<string, mixed>
-     * }|null
+     * }
      */
-    public function getContext(string $name): ?array
+    public function getContext(string $name): array
```

Comment:
This is the most notable change, I tried to always return an array with the context here. More like we did on the admin bundle. you have a hasContext method, and if you want to be sure you don't run into exception, you need to call it before getContext.

Reply:
It could be nice to add the deprecation in 3.x

Reply:
To not block this PR I have added this issue: https://github.com/sonata-project/SonataMediaBundle/issues/2014